core/src/OpenMP/Kokkos_OpenMP.hpp

```diff
@@ -154,9 +157,11 @@ inline int OpenMP::impl_thread_pool_rank() noexcept {
   KOKKOS_IF_ON_DEVICE((return -1;))
 }
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
 inline bool OpenMP::is_asynchronous(OpenMP const& /*instance*/) noexcept {
```

Comment:
I can never remember if we need to repeat the deprecation attribute on the definition.
I think it might be worth inlining here.